pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -250,6 +254,42 @@ func (s *SSOSettingsService) encryptSecrets(ctx context.Context, settings map[st
 	return result, nil
 }
 
+// initReload starts a background process for reloading the SSO settings for all providers at a fixed interval
+// it is useful for high availability setups running multiple Grafana instances
+func (s *SSOSettingsService) initReload() {
```

Comment:
Sorry did not think of this earlier but we should probably register this as a background service. Then this would gracefully terminate if grafana stops.

What you need to do is register this service [here](https://github.com/grafana/grafana/blob/main/pkg/registry/backgroundsvcs/background_services.go).

You can look at https://github.com/grafana/grafana/blob/main/pkg/services/anonymous/anonimpl/impl.go#L172 for how this is usually implemented

Reply:
Done.